### DISCUSSION TITLE
Poor error message when many field not set to True when passed a list of instances

### DISCUSSION THREAD

**Discussion by justin107d** [encode/django-rest-framework#8711] [Ideas & Suggestions]
It would be nice if when multiple instances are passed to the serializer that it told you the error was because `many=True` was not set.  

This happened on a ModelSerializer with certain fields excluded shortly after adding a field to the model.  The error received referred to the 'missing' field instead of receiving multiple instances.  Setting `many=True` fixed the issue.

I am new to open source and would like to help if I can or if it is appropriate here.

I think adding something similar to the following around line 117-118 of the serializer.py

```
from django.db.models.query import QuerySet
...
if isinstance(instance, QuerySet) and kwargs.get('many', None) != True:
    raise TypeError('Queryset has multiple instances.  Try submitting only one instance or set many=True')
```

**Comment by lovelydinosaur** [comment]
Interesting idea. Could be worth a go yup.
If you're up for submitting a pull request for this we could review and go from there.

**Reply by justin107d** [reply]
Thanks Tom.  I submitted a request with commit id:  63981eca8111faa773792cf5247cdf2f299e7fb2
